src/Raven.Server/Documents/Indexes/IndexField.cs

```diff
@@ -44,12 +44,13 @@ public IndexField()
             Storage = FieldStorage.No;
         }
 
-        public static IndexField Create(string name, IndexFieldOptions options, IndexFieldOptions allFields)
+        public static IndexField Create(string name, IndexFieldOptions options, IndexFieldOptions allFields, int id = 0)
         {
             var field = new IndexField
             {
                 Name = name,
-                Analyzer = options.Analyzer ?? allFields?.Analyzer
+                Analyzer = options.Analyzer ?? allFields?.Analyzer,
+                Id = id
```

Comment:
Something to note, we need to handle dynamic fields. That is something that we haven't had to deal with so far